Added "COMMAND" and "FAIL_URL"

dantebarba opened the pull request:
+ "COMMAND": variable was added to parametrize rclone execution. Now is possible to use not only sync command, but move, copy, or delete.
+ "FAIL_URL": Allows the user to set an URL for failure notification. With the standard "CHECK_URL", the script won't notify to the user when the command ends badly. The user would only be alerted when the service is down for a specific interval of time. Now the user is able to receive notifications when the rclone command did not end successfully.  The default FAIL_URL is the "CHECK_URL" with an appended path "/fail", as defined by [healthchecks.io](https://healthchecks.io/docs/). This url can be overridden by the user setting the "FAIL_URL" env var.

dantebarba committed:
FAIL_URL and parametrized commad

- Added "$COMMAND" variable to specify the desired command to execute
- Added "$FAIL_URL" variable to specify the failure default URL in case of command failure.

dantebarba committed:
removed unnecesary env var

dantebarba committed:
removed unnecesary env

dantebarba committed:
readme formatting

pfidr34 committed:
update to latest version

pfidr34 committed:
Update LICENSE.md

pfidr34 committed:
Update README.md

pfidr34 committed:
Update README.md

bcardiff commented on README.md:
```diff
@@ -39,12 +39,15 @@ $ docker run --rm -it -v $(pwd)/config:/config bcardiff/rclone
 * `CRON_ABORT` crontab schedule `0 6 * * *` to abort sync at 6am
 * `FORCE_SYNC` set variable to perform a sync upon boot
 * `CHECK_URL` [healthchecks.io](https://healthchecks.io) url or similar cron monitoring to perform a `GET` after a successful sync
+*  `FAIL_URL` Fail URL to perform a `GET` after unsuccesful execution. Default: `CHECK_URL` with appended "/fail" at the end
```
```suggestion
* `FAIL_URL` Fail URL to perform a `GET` after unsuccessful execution. Default: `CHECK_URL` with appended "/fail" at the end
```

bcardiff replied:
Just a tiny typo / extra space and is ready to go. I didn't know about `/fail`

bcardiff commented:
There is a message "WARNING: A previous sync is still running. Skipping new sync command." that might need to be updated also if the command is not sync probably

pfidr34 committed:
Update Dockerfile

pfidr34 committed:
Merge branch 'master' of https://github.com/pfidr34/docker-rclone

dantebarba commented:
@bcardiff I think @pfidr34's version is better than this one, I would suggest to merge his pull request adding the "/fail" part from this pull request.

bcardiff committed:
Update README.md

typo

Co-Authored-By: dantebarba <dantebarba@gmail.com>

dantebarba committed:
Merge branch 'master' of https://github.com/bcardiff/docker-rclone

* 'master' of https://github.com/bcardiff/docker-rclone:
  Fix shellcheck failure
  Upgrade base alpine image and allow image to be parametrized
  Smoke test that rclone binary is available
  Clean up test docker images
  Fix #3. Avoid recursion when checking for empty source
  Fix deprecation warning due to MAINTAINER
  Add specs. Fix custom versions builds.
  Make ARCH and RCLONE_VERSION build-time arguments (#5)

dantebarba committed:
Merge branch 'master' of https://github.com/dantebarba/docker-rclone

* 'master' of https://github.com/dantebarba/docker-rclone:
  Update README.md

dantebarba committed:
Added travis ci badge

pfidr34 committed:
Create FAIL_URL handling proposed by @dantebarba

pfidr34 commented:
> @bcardiff I think @pfidr34's version is better than this one, I would suggest to merge his pull request adding the "/fail" part from this pull request.

Thanks @dantebarba I have added your /fail code with a couple tweaks.

pfidr34 committed:
Add 'move' as a acceptable rclone operation.

pfidr34 committed:
typo

pfidr34 committed:
Add ability to roate logs after x days

pfidr34 committed:
Change variable name

pfidr34 committed:
Autobuild for Rclone 1.48

tlex committed:
Simplifies `apk add`

* removes the option `-U` (`Maximum AGE (in minutes) for index in cache before refresh` set to 1)
* removes the deletion of the local cache
* adds the option `--no-cache` (`Do not use any local cache path`)

pfidr34 committed:
Merge pull request #1 from tlex/simplify-apk-add

Simplifies `apk add`

pfidr34 committed:
Fix log rotation

When there are no logs to delete using the LOG_ROTATE variable will cause the container to crash.

Thanks @Skaronator

pfidr34 committed:
Add change log to readme.md

pfidr34 committed:
Healthchecks.io enhancements

Send start command prior to sync
Add debug messages to healthcheck.io calls

pfidr34 committed:
Update README.md

pfidr34 committed:
Update sync.sh

Fix for log rotation

pfidr34 committed:
Change log directory check

Commit:
Add env variable SYNC_OPTS_EVAL.

* SYNC_OPTS_EVAL is a interpolated string like
'--backup-dir  /old`date -I`'.
The commands are first evaluated like in a normal shell, e.g. to
'--backup-dir  /old2019-09-12' and then appended to SYNC_OPTS.
* This is useful to to make iterative backups and keep the old versions of the file in the
backup-dir.

Commit:
Add documentation for SYNC_OPTS_EVAL.

Commit:
Fix typos in README.

pfidr34 committed:
Merge pull request #5 from mistermult/eval-args

Add environment variable SYNC_OPTS_EVAL.

pfidr34 committed:
Update SYNC_EVAL logic

Only evaluate and print sync eval if the option was provided.

pfidr34 committed:
Update README.MD

pfidr34 committed:
Add ability to sync once then exit

pfidr34 committed:
Update README.md

pfidr34 committed:
Update README.md

pfidr34 committed:
Update README.md

pfidr34 committed:
Increment version

pfidr34 committed:
Version update

pfidr34 committed:
Update version

Commit:
Initial support for passing UID/GID

Commit:
typo

Commit:
Don't assume root, and switch to user if they arleady exist

pfidr34 committed:
Merge pull request #10 from mikeburgh/master

Assign UID and GID

pfidr34 committed:
Update Readme

pfidr34 committed:
Update version

JakeWharton committed:
Use correct user for Docker Hub image

pfidr34 committed:
Merge pull request #11 from JakeWharton/patch-2

Use correct user for Docker Hub image

pfidr34 committed:
Update rclone version

pfidr34 committed:
Handle spaces in SYNC_SRC and SYNC_DEST options

pfidr34 committed:
Change how sync.sh handles errors from rclone

pfidr34 committed:
Remove DEBUG echo

pfidr34 committed:
Various updates, see README.md

pfidr34 committed:
Updated to Rclone 1.52.0 and added build script

pfidr34 committed:
eval rclone command | set error handling on source dir check

pfidr34 committed:
Fix whitespace issues in SYNC_SRC and SYNC_DEST

pfidr34 committed:
Revert last change

dantebarba committed:
remove sync warning

pfidr34 committed:
Update to Rclone 1.52.1

pfidr34 committed:
Update to 1.52.2

pfidr34 committed:
Update to 1.52.3

pfidr34 committed:
Add ability to send log data to healthchecks.io

pfidr34 committed:
Update to 1.53.0

pfidr34 committed:
Update to 1.53.1

pfidr34 committed:
Update to 1.53.2

pfidr34 committed:
Update to 1.53.3

pfidr34 committed:
Update to 1.53.4

pfidr34 committed:
Update to 1.54.0

pfidr34 committed:
Update to 1.54.1

pfidr34 committed:
Update to 1.55.0 & 1.55.1

pfidr34 committed:
Update to 1.56.0

pfidr34 committed:
Update to 1.56.1

pfidr34 committed:
Update to 1.56.2

pfidr34 committed:
Update to 1.57.0

mistermult committed:
Fix wrong healtcheck for DIR_CHECK, no OUTPUT_LOG.

- This fixes the problem that the reported healtcheck is ok despite rclone exiting with an error, i.e. with return code unequal from 0.
- Moreover, it prints the return code for easy inspection of logs.

pfidr34 committed:
Merge pull request #29 from mistermult/fix-return-code

Critical: Fix healtcheck reporting ok but rclone failed in case DIR_CHECK, no OUTPUT_LOG.

pfidr34 committed:
Update to 1.58.0

pfidr34 committed:
Update to 1.59.0

pfidr34 committed:
Update to 1.61.1

pfidr34 committed:
Update to version 1.63.0

dantebarba committed:
Merge remote-tracking branch 'fork/master' into pfidr34/docker-rclone

dantebarba committed:
Merge pull request #3 from dantebarba/pfidr34/docker-rclone

Update my docker-rclone fork with pfidr34 fork because bcardiff is outdated and not maintained.

dantebarba committed:
Added new SUCCESS_CODES configuration

+ Success codes are configured by the "SUCCESS_CODES" env var
+ By default the value is "0"
+ Other codes can be specified
List of exit codes
0 - success
1 - Syntax or usage error
2 - Error not otherwise categorised
3 - Directory not found
4 - File not found
5 - Temporary error (one that more retries might fix) (Retry errors)
6 - Less serious errors (like 461 errors from dropbox) (NoRetry errors)
7 - Fatal error (one that more retries won't fix, like account suspended) (Fatal errors)
8 - Transfer exceeded - limit set by --max-transfer reached
9 - Operation successful, but no files transferred

dantebarba committed:
Changed the incompatible bash operator with a compatible sh solution

dantebarba committed:
Solve unexpected "then"

dantebarba committed:
Merge pull request #4 from dantebarba/feat/add-success-codes

Added new SUCCESS_CODES configuration
